docs/modules/ROOT/pages/advanced-how-tos/managing-multiple-versions.adoc

```diff
@@ -161,6 +161,27 @@ event == "push" && target_branch == "v1.0.0"
 
 * The `appstudio.openshift.io/application` and `appstudio.openshift.io/component` labels must be adjusted to specify the right Application and Component respectively. Failing to do this will cause builds of the pipeline to be associated with the wrong application or component.
 
+=== Create ImageRepositories for each component
```

Comment:
It must be done by UI automatically on Component creation.

Reply:
Added

Reply:
@mmorhun can we use the `image.redhat.com/image` annotation on the component instead of defining this resource?  What happens if we define the same annotation value on multiple components? Notably when I was testing this, everything seems to have worked fine while all components had the same `output-image` value configure for the pipeline, and only one of them had the annotation defined on it.

What is the criteria that decides when the push secret is made available to a pipeline? Is this by matching component and application names? Does this work differently when having the `ImageRepository` resource v.s. the `image.redhat.com/image` annotation?

Reply:
No, we cannot use `image` annotation because it's like a status: it just shows current state. Moreover, it's deprecated and we are on removing it now. `ImageRepository` is the way to go. It has wider functionality and clearer interface.

Reply:
> What is the criteria that decides when the push secret is made available to a pipeline?

It should be linked to the pipeline service account